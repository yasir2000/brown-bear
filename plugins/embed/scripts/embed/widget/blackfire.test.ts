/*
 * Copyright (c) Enalean, 2020 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { insertBlackfire } from "./blackfire";

describe("blackfire", () => {
    describe("insertBlackfire", () => {
        let doc: Document;
        beforeEach(() => {
            doc = createLocalDocument();
        });

        it("insert an iframe", () => {
            const a = doc.createElement("a");
            a.href = "https://blackfire-url.example.com/graph";
            a.innerText = "blackfire";
            doc.body.appendChild(a);

            insertBlackfire(a);

            expect(doc.body).toMatchSnapshot();
        });
    });
});

function createLocalDocument(): Document {
    return document.implementation.createHTMLDocument();
}
