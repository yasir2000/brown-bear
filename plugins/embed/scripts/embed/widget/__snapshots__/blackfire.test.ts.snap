// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blackfire insertBlackfire insert an iframe 1`] = `
<body>
  <a
    href="https://blackfire-url.example.com/graph"
  />
  <br />
  <iframe
    allowfullscreen=""
    height="650"
    referrerpolicy="same-origin"
    sandbox="allow-scripts allow-same-origin"
    src="https://blackfire-url.example.com/embed"
    width="100%"
  />
</body>
`;
